src/appkit.cs

```diff
@@ -23721,4 +24387,592 @@ interface NSMenuValidation
 		[Export ("validateMenuItem:")]
 		bool ValidateMenuItem (NSMenuItem menuItem);
 	}
+
+	public interface INSCandidateListTouchBarItemDelegate {}
+
+	[Mac (10,12,1)]
+	[BaseType (typeof(NSTouchBarItem))]
+	interface NSCandidateListTouchBarItem
+	{
+		[Export ("initWithIdentifier:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string identifier);
+
+		[NullAllowed, Export ("client", ArgumentSemantic.Weak)]
+		INSTextInputClient Client { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		INSCandidateListTouchBarItemDelegate Delegate { get; set; }
+
+		[Export ("collapsed")]
+		bool Collapsed { [Bind ("isCollapsed")] get; set; }
+
+		[Export ("allowsCollapsing")]
+		bool AllowsCollapsing { get; set; }
+
+		[Export ("candidateListVisible")]
+		bool CandidateListVisible { [Bind ("isCandidateListVisible")] get; }
+
+		[Export ("updateWithInsertionPointVisibility:")]
+		void UpdateWithInsertionPointVisibility (bool isVisible);
+
+		[Export ("allowsTextInputContextCandidates")]
+		bool AllowsTextInputContextCandidates { get; set; }
+
+		[NullAllowed, Export ("attributedStringForCandidate", ArgumentSemantic.Copy)]
+		Func<NSObject, nint, NSAttributedString> AttributedStringForCandidate { get; set; }
+
+		[Export ("candidates", ArgumentSemantic.Copy)]
+		NSObject[] Candidates { get; }
```

Comment:
There are a large number of Candidate APIs here and they all are marked as NSObject. I could go debug and find what they return, but since it's already made swift happy, I'd assume they'd use a more specific type...